README.md

```diff
@@ -1,66 +1,117 @@
 
-In other words, RAG is essentially a guess as to what the LLM might need to generate a good answer and loading it into the context. Because it relies on text or vector search, it only works effectively for use cases where the user question can be easily mapped to a good retrieval query.
-And even for use cases where RAG works reasonably well - such as semantic search - it can be very inefficient. For example, if a user asks to "find all information on the 'nightfall' project from last December" RAG may find information on the project but has no way to limit it to the given timeframe. 
+## Why Acorn Agent?
 
-apiRAG does not suffer from these shortcomings because it relies on the LLM to determine what information it needs and relies on the intelligence of LLMs to translate user queries into efficient and relevant retrieval requests.
+Most agent frameworks are either too complex with lots of connectors, configurations, and options or abstract too much away from the actual LLM calls which doesn't give you enough control to fine-tune and cost-optimize your agents.
 
-## How does apiRAG Work?
+We wanted to build an agent framework that makes it easy to get started and experiment with different models but doesn't leave you hanging when you need low-level control to optimize performance and cost. In addition, we wanted an agent framework that automatically benefits from the rapid innovation on LLMs without requiring rewriting your agent.
 
-![Diagram of how apiRAG executes user requests](img/apiRAG-diagram.png)
+* Acorn Agent is simple: It manages and invokes tools for LLMs efficiently and safely. That's it. [We believe tools are all you need](#tools-are-all-you-need) to succeed with LLMs. And it future-proofs your agent as LLMs get better and better at using tools.
+* Acorn Agent is flexible: you can use the lightweight abstraction layers that Acorn agent provides to get started quickly and swap out models easily, but you can also use the tooling framework with any model client library or model API for full control over each model invocation.
 
-At it's core, apiRAG is a configuration format that defines a set of LLM functions and how they map to API queries.
-It extends OpenAI's function call configuration with additional information on how to execute the function against an existing API.
+## Acorn Agent Features
 
-For more information on apiRAG's configuration format, check out the [format documentation](FORMAT.md).
+* On-Demand Data Retrieval from GraphQL APIs, REST APIs, or JDBC-compatible databases.
+* Pluggable models and model providers like OpenAI (GPT models), Bedrock (Llama, and others), Google Vertex (Gemini), and Groq (Llama, Mixtral, and others).
+* Message history to preserve context between agent interactions.
+* Custom data visualizations.
+* Easy integration with Spring Boot and other web development frameworks with in-code or file-based configuration.
+* UI integration through function callbacks.
```

Comment:
@mbroecheler What do you mean by this point? Function Callbacks is a bit confusing here when we talk about a function calling platform.

Reply:
clarified.